internal/resources/metal/vlan/models.go

```diff
@@ -21,29 +21,34 @@ type DataSourceModel struct {
 	AssignedDevicesIds types.List   `tfsdk:"assigned_devices_ids"`
 }
 
-func (m *DataSourceModel) parse(vlan *packngo.VirtualNetwork) (d diag.Diagnostics) {
-	m.ID = types.StringValue(vlan.ID)
-	m.VlanID = types.StringValue(vlan.ID)
-	m.Vxlan = types.Int64Value(int64(vlan.VXLAN))
+func (m *DataSourceModel) parse(vlan *metalv1.VirtualNetwork) (d diag.Diagnostics) {
+	m.ID = types.StringValue(vlan.GetId())
+	m.VlanID = types.StringValue(vlan.GetId())
+	m.Vxlan = types.Int64Value(int64(vlan.GetVxlan()))
 	m.Facility = types.StringValue("")
 
-	if vlan.Description != "" {
-		m.Description = types.StringValue(vlan.Description)
+	if vlan.GetDescription() != "" {
+		m.Description = types.StringValue(vlan.GetDescription())
 	}
 
-	if vlan.Project.ID != "" {
-		m.ProjectID = types.StringValue(vlan.Project.ID)
+	if vlan.AdditionalProperties["project"] != nil {
```

Comment:
This will require a change to the OpenAPI spec to declare the `project` attribute for VLANs

Reply:
On closer inspection the `Project` field in `packngo` came from the `assigned_to` attribute in the API response.  That attribute is defined in the API spec and is available via `equinix-sdk-go`.